### PR TITLE
Set Default User Locale

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/SQSERVICES-1097

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.
   - [x] If new config options introduced: added usage description under docs/reference/config-options.md
   - [x] If new config options introduced: recommended measures to be taken by on-premise instance operators.

### CODE REVIEW

File: services/brig/src/Brig/Options.hs [hunk 1]
```diff
@@ -439,9 +439,10 @@ data Settings = Settings
     --   field names and values), should be in sync
     --   with Spar
     setRichInfoLimit :: !Int,
-    -- | Default locale to use
-    --   (e.g. when selecting templates)
-    setDefaultLocale :: !Locale,
+    -- | Default locale to use when selecting templates
+    setDefaultTemplateLocale :: !Locale,
+    -- | Default locale to use for users
+    setDefaultUserLocale :: !Locale,
```
Comment (fisx) [inline services/brig/src/Brig/Options.hs:445]: I like the names you picked.

Haven't checked the context, so some or all of these may already be taken care of.   (Maybe the haddocks could be more explicit about things, and also explain what the difference is between the two?)

- In order to make upgrades for on-prem customers easier, you should only add optional config fields if possible, so make sure this parses if either or both are missing.
- Make the actual value (hard-wired default or not) of `setDefaultTemplateLocale` the default value of `setDefaultUserLocale`, and
- `en` the default for `setDefaultTemplateLocale`.
- Rationale for the last three points: you're using `setDefaultUserLocale` a lot in lieu of the deleted `setDefaultLocale`, the latter had a hard-wired default of `en`, and was primarily used for `setDefaultTemplateLocale`.

**Review by fisx — APPROVED** [review]
Nice, especially the LOC stats!  :)

Yes, this could work.  Please test as you suggested before merging, though.

File: changelog.d/2-features/helm-brig-setDefaultLocale [hunk 2]
```diff
@@ -1 +1 @@
-Allow configuring setDefaultLocale in brig using helm chart
\ No newline at end of file
+Allow configuring setDefaultLocale in brig using helm chart
```
Comment (fisx) [inline changelog.d/2-features/helm-brig-setDefaultLocale:1]: get rid of that extra newline, too?

File: changelog.d/2-features/pr-2028 [hunk 3]
```diff
@@ -0,0 +1 @@
+Renamed the brig config `setDefaultLocale` to `setDefaultUserLocale` and introduced a new config `setDefaultTemplateConfig`.
```
Comment (fisx) [inline changelog.d/2-features/pr-2028:1]:
```suggestion
Renamed the brig config `setDefaultLocale` to `setDefaultUserLocale` and introduced a new config `setDefaultTemplateLocale`.
```

Comment (fisx) [reply changelog.d/2-features/pr-2028:1]: Or:

```suggestion
Specialize setDefaultLocale to distinguish between default user preference and default template if the one in the preferred language is n/a.
```

**Review by fisx — COMMENTED** [review]
I'm assuming you tested the `DefaultUserLocale` trick manually?  Then except for the missing details in the haddocks this looks very shippable.

File: services/brig/src/Brig/Options.hs [hunk 4]
```diff
@@ -504,11 +504,11 @@ data Settings = Settings
...
 
-defaultUserLocale :: Locale
-defaultUserLocale = Locale (Language EN) Nothing
-
 defaultTemplateLocale :: Locale
-defaultTemplateLocale = defaultUserLocale
+defaultTemplateLocale = Locale (Language EN) Nothing
+
+defaultUserLocale :: Locale
+defaultUserLocale = defaultUserLocale
```
Comment (fisx) [inline services/brig/src/Brig/Options.hs:511]: cycle.

```suggestion
defaultUserLocale = defaultTemplateLocale
```

(anyway it'd also be ok with me if you made both of them just `en`.)

File: docs/reference/config-options.md [hunk 5]
```diff
@@ -319,3 +319,27 @@ When a `null` value is encountered, it is assumed to be
 `defaultForNull`.
 
 (Introduced in https://github.com/wireapp/wire-server/pull/1811.)
+
+### Locale
+
+The brig server config option `setDefaultLocale` has been replaced by `setDefaultUserLocale` and `setDefaultTemplateLocale`. Both settings are optional and `setDefaultTemplateLocale` defaults to `EN` and `setDefaultLocale` defaults to `setDefaultTemplateLocale`. If `setDefaultLocale` was not set or set to `EN` before this release, nothing needs to be done. If `setDefaultLocale` was set to any other language other than `EN` the name of the setting should be changed to `setDefaultTemplateLocale`.
```
Comment (fisx) [inline docs/reference/config-options.md:325]:
```suggestion

#### setDefaultLocale (deprecated / ignored)

The brig server config option `setDefaultLocale` has been replaced by `setDefaultUserLocale` and `setDefaultTemplateLocale`. Both settings are optional and `setDefaultTemplateLocale` defaults to `EN` and `setDefaultLocale` defaults to `setDefaultTemplateLocale`. If `setDefaultLocale` was not set or set to `EN` before this change, nothing needs to be done. If `setDefaultLocale` was set to any other language other than `EN` the name of the setting should be changed to `setDefaultTemplateLocale`. 
```

File: changelog.d/0-release-notes/pr-2028 [hunk 6]
```diff
@@ -0,0 +1 @@
+The brig server config option `setDefaultLocale` has been replaced by `setDefaultUserLocale` and `setDefaultTemplateLocale`. Both settings are optional and `setDefaultTemplateLocale` defaults to `EN` and `setDefaultLocale` defaults to `setDefaultTemplateLocale`. If `setDefaultLocale` was not set or set to `EN` before this release, nothing needs to be done. If `setDefaultLocale` was set to any other language other than `EN` the name of the setting should be changed to `setDefaultTemplateLocale`.
```
Comment (fisx) [inline changelog.d/0-release-notes/pr-2028:1]:
```suggestion
The brig server config option `setDefaultLocale` has been replaced by `setDefaultUserLocale` and `setDefaultTemplateLocale` (see docs/reference/config-options.md for details)
```

**Review by fisx — APPROVED** [review]
NB the suggestion to fix the cycle (black hole), otherwise LGTM!

File: docs/reference/config-options.md [hunk 7]
```diff
@@ -319,3 +319,30 @@ When a `null` value is encountered, it is assumed to be
...
 (Introduced in https://github.com/wireapp/wire-server/pull/1811.)
+
+### Locale
+
+
+#### setDefaultLocale (deprecated / ignored)
+
+The brig server config option `setDefaultLocale` has been replaced by `setDefaultUserLocale` and `setDefaultTemplateLocale`. Both settings are optional and `setDefaultTemplateLocale` defaults to `EN` and `setDefaultLocale` defaults to `setDefaultTemplateLocale`. If `setDefaultLocale` was not set or set to `EN` before this change, nothing needs to be done. If `setDefaultLocale` was set to any other language other than `EN` the name of the setting should be changed to `setDefaultTemplateLocale`. 
+
+#### `setDefaultTemplateLocale`
```
Comment (jschaul) [inline docs/reference/config-options.md:330]: Which templates are actually complete in wire-server other than EN? i.e. this settings can't be set to anything other than EN at this point in time unless more templates are provided inside wire-server's brig/deb/templates folder, right? Perhaps the description here should make that clearer.

**Pull request merged**